beacon-chain/db/kv/genesis_test.go

```diff
@@ -28,41 +29,70 @@ func testGenesisDataSaved(t *testing.T, db iface.Database) {
 	ctx := context.Background()
 
 	gb, err := db.GenesisBlock(ctx)
-	assert.NoError(t, err)
-	assert.NotNil(t, gb)
+	require.NoError(t, err)
+	require.NotNil(t, gb)
 
 	gbHTR, err := gb.Block().HashTreeRoot()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	gss, err := db.StateSummary(ctx, gbHTR)
-	assert.NoError(t, err)
-	assert.NotNil(t, gss)
+	require.NoError(t, err)
+	require.NotNil(t, gss)
 
 	head, err := db.HeadBlock(ctx)
-	assert.NoError(t, err)
-	assert.NotNil(t, head)
+	require.NoError(t, err)
+	require.NotNil(t, head)
 
 	headHTR, err := head.Block().HashTreeRoot()
-	assert.NoError(t, err)
-	assert.Equal(t, gbHTR, headHTR, "head block does not match genesis block")
+	require.NoError(t, err)
+	require.Equal(t, gbHTR, headHTR, "head block does not match genesis block")
 }
 
 func TestLoadGenesisFromFile(t *testing.T) {
+	// for this test to work, we need the active config to have these properties:
+	// - fork version schedule that matches mainnnet.genesis.ssz
+	// - name that does not match params.MainnetName - otherwise we'll trigger the codepath that loads the state
+	//   from the compiled binary.
+	// to do that, first we need to rewrite the mainnet fork schedule so it won't conflict with a renamed config that
+	// uses the mainnet fork schedule. construct the differently named mainnet config and set it active.
+	// finally, revert all this at the end of the test.
+
+	// first get the real mainnet out of the way by overwriting it schedule.
+	cfg, err := params.Registry.GetByName(params.MainnetName)
+	require.NoError(t, err)
+	cfg = cfg.Copy()
+	reversioned := cfg.Copy()
+	params.FillTestVersions(reversioned, 127)
+	undo, err := params.Registry.SetActiveWithUndo(reversioned)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, undo())
+	}()
+
+	// then set up a new config, which uses the real mainnet schedule, and activate it
+	cfg.ConfigName = "genesis-test"
+	undo2, err := params.Registry.SetActiveWithUndo(cfg)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, undo2())
+	}()
```

Comment:
note that defer is LIFO, so we'll roll back to the config set on line 66 before we revert all the way back to clean pre-test state.